src/bin/analyzer.js

```diff
@@ -81,10 +86,33 @@ let {
   logLevel,
   open: openBrowser,
   exclude: excludeAssets,
+  info,
   args: [bundleStatsFile, bundleDir]
 } = program;
 const logger = new Logger(logLevel);
 
+const packageInfo = {
+  node: process.version,
+  npm: 'npm -v',
+  yarn: 'yarn -v',
+  webpack: 'webpack-cli -v',
+  webpackBundleAnalyzer: require('../../package.json').version
+};
+
+if (info) {
+  console.log(' Environment info:-');
+  try {
+    Object.keys(packageInfo).forEach((pkg, index) => {
+      [0, 4].includes(index) ?
```

Comment:
This is just a temporary (weird) implementation just to make the `exec` version work as intended.

Reply:
Could you explain why we need this temporary implementation if it won't be merged like that anyway? Is this PR still WIP?

Reply:
Can you just let me know the implementation strategy that you've in mind :thinking:

Reply:
```js
async function printInfo() {
  const packages = {
    npm: {
      command: 'npm -v',
      processOutput: output => `v${output.trim()}`
    },
    node: process.version
  };

  const versionsPromises = Object.entries(packages).map(([name, versionData]) => {
    let promise;
    if (typeof versionData === 'string') {
      promise = Promise.resolve(versionData);
    } else {
      promise = exec(versionData.command)
        .then(result => result.stdout)
        .then(output => (versionData.processOutput ? versionData.processOutput(output) : output))
        .catch(() => {});
    }

    return promise.then(version => [name, version]);;
  });

  for (const promise of versionsPromises) {
    // Printing results in the order
    const [name, version] = await promise;
    console.log(`${name}: ${version}`);
  }
}

printInfo().then(() => process.exit());
```
In this case all commands will be executed in parallel, but printed in the right order.

Reply:
Since we're having multiple `if` blocks the `help text` fires up without waiting for the `Promise` to resolve.